src/Storages/HivePartitioningUtils.cpp

```diff
@@ -69,7 +69,11 @@ NamesAndTypesList extractHivePartitionColumnsFromPath(
 
     const auto hive_map = parseHivePartitioningKeysAndValues(sample_path);
 
+    std::map<std::string_view, std::string_view> hive_map_sorted;
```

Comment:
I think you should actually just change the `extractKeyValuePair` data structure here https://github.com/ClickHouse/ClickHouse/blob/3e9b687b53e70e91e6500e65373e71aeb9c09320/src/Functions/keyvaluepair/impl/StateHandlerImpl.h#L640. 

For now, this is the only usage of this function, so I think it is ok to make that change. Wdyt?